### PR TITLE
Add discovery.nerve component and converter

### LINKED ISSUE #2277: Flow: discovery.nerve component

### PR DESCRIPTION
**Pull request by tpaschalis**
#### Which issue(s) this PR fixes
Closes #2277 

#### PR Checklist

- [X] CHANGELOG.md updated
- [X] Documentation added
- [X] Tests updated
- [X] Config converters updated

### CODE REVIEW

**Comment by thampiotr** [comment]
Similar to this comment: https://github.com/grafana/agent/pull/5036#issuecomment-1702871349
there is a code path using which people could use nerve SD in promtail and the conversion is right now not supported?

**Review by clayton-cornell — COMMENTED** [review]
No docs suggestions. Good enough as-is.

File: docs/sources/flow/reference/components/discovery.nerve.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Arguments
+
+The following arguments are supported:
+
+Name               | Type           | Description                          | Default       | Required
+------------------ | -------------- | ------------------------------------ | ------------- | --------
+`servers`          | `list(string)` | The Zookeeper servers.               |               | yes
+`paths`            | `list(string)` | The paths to look for targets at.    |               | yes
+`timeout`          | `duration`     | The timeout to use.                  | `"10s"`       | no
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.nerve.md:28]:
```suggestion
`paths`            | `list(string)` | The target paths.    |               | yes
`timeout`          | `duration`     | The timeout.                  | `"10s"`       | no
```

We could simplify these descriptions.

**Review by clayton-cornell — COMMENTED** [review]
Minor doc suggestion

File: docs/sources/flow/reference/components/discovery.nerve.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+`discovery.nerve` does not expose any component-specific debug metrics.
+
+## Example
+
+```river
+discovery.nerve "example" {
+	servers = ["localhost"]
+	paths   = ["/monitoring"]
+    timeout = "1m"
```
Comment (wildum) [inline docs/sources/flow/reference/components/discovery.nerve.md:74]: nit: the indent seems incorrect here

**Review by wildum — APPROVED** [review]
LGTM

**Pull request merged**